src/server/command_registry.cc

```diff
@@ -60,18 +60,18 @@ void CommandId::Invoke(CmdArgList args, ConnectionContext* cntx) const {
   ent.second += (after - before) / 1000;
 }
 
-optional<facade::ErrorReply> CommandId::Validate(CmdArgList args) const {
-  if ((arity() > 0 && args.size() != size_t(arity())) ||
-      (arity() < 0 && args.size() < size_t(-arity()))) {
+optional<facade::ErrorReply> CommandId::Validate(CmdArgList tail_args) const {
+  if ((arity() > 0 && tail_args.size() + 1 != size_t(arity())) ||
+      (arity() < 0 && tail_args.size() + 1 < size_t(-arity()))) {
     return facade::ErrorReply{facade::WrongNumArgsError(name()), kSyntaxErrType};
   }
 
-  if (key_arg_step() == 2 && (args.size() % 2) == 0) {
+  if (key_arg_step() == 2 && (tail_args.size() % 2) != 0) {
```

Comment:
![image](https://github.com/dragonflydb/dragonfly/assets/20553775/3c76bb01-2094-41b1-9b63-dec60e6a77a1)

Theoretically yes, if all commands consisted only of keys/values. Other parameters, including optional, break the rule. It just happens so that k/v paired commands don't have any optional arguments and params, so we check that the arguments are paired

Reply:
lol, gotcha :)